src/passes/OptimizeInstructions.cpp

```diff
@@ -2404,17 +2404,53 @@ struct OptimizeInstructions
         Bits::getMaxBits(curr->ifFalse, this) <= 1) {
       // The condition and both arms are i32 booleans, which allows us to do
       // boolean optimizations.
-      Expression* x;
-      Expression* y;
 
       // x ? y : 0   ==>   x & y
-      if (matches(curr, select(any(&y), ival(0), any(&x)))) {
```

Comment:
Starting the match with `select` even though we already know it is select is intentional so we can read the whole pattern directly without any context. It also lets us use variable names in the output that directly match the names in the input pattern comment.

Reply:
Hmm, interesting. That is less efficient potentially but maybe it is more readable...